### PR TITLE
chore(font-face): @font-face mixin

### PR DESCRIPTION
**Pull request by bhough**
@font-face mixin, accepts both file-based and local references. Also is configurable for file formats, font options, and unicode-range.

Addresses an item in #1

### CODE REVIEW

**Comment by bhough** [comment]
@mxstbr this is ready for review. Travis is timing out on tests only on the push build, but I think it is a hiccup on their end. Especially want your feedback on the examples, since @font-face works differently than traditional CSS properties.

File: src/mixins/fontFace.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      'unicode-range': unicodeRange,
+      'font-stretch': fontStretch,
+      'font-style': fontStyle,
+      'font-variant': fontVariant,
+      'font-weight': fontWeight,
+    },
+  }
+
+  // Removes undefined fields for cleaner css object.
+  return JSON.parse(JSON.stringify(fontFaceDeclaration))
```
Comment (mxstbr) [inline src/mixins/fontFace.js]: Is there a faster way of doing this?

**Comment by bhough** [comment]
@mxstbr I looked at pruning by looping over with Object.keys and it proved to be about the same speed, with more code. The question is if we are okay with a slightly more opaque approach with a comment vs a more verbose but slightly clear approach (with Object.keys).

File: src/mixins/fontFace.js (hunk 1 above)
Comment (bhough) [inline src/mixins/fontFace.js]: @mxstbr I looked at pruning by looping over with Object.keys and it proved to be about the same speed, with more code. The question is if we are okay with a slightly more opaque approach with a comment vs a more verbose but slightly clear approach (with Object.keys).

**Pull request merged**